tools/codegen/model_generator/model.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model_generator
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	tmplExtns "istio.io/mixer/tools/codegen/template_extension"
```

Comment:
nit: move `tmplExtns` to separate block (as this is an istio.io import). I also suggest just `tmpl`.